src/main/java/org/plumelib/bcelutil/StackMapUtils.java

```diff
@@ -0,0 +1,1205 @@
+          mgen.addLocalVariable(
+              "DaIkOnTeMp" + offset, live_type, offset, il.findHandle(live_start), null);
+      debug_instrument.log(
+          "Added local  %s, %d, %d : %s, %s%n",
+          new_lvg.getIndex(),
+          new_lvg.getStart().getPosition(),
+          il.getEnd().getPosition(),
+          new_lvg.getName(),
+          new_lvg.getType());
+      min_size = Math.min(min_size, live_type.getSize());
+    } else {
+      if (min_size == 3) {
+        // did not find a temp in any of the stack maps; that must mean the
+        // temp live range is in between two stack maps or after last stack map
+        // need to scan all byte codes to get live range and type
+        // HACK: for now, just make a guess and put it after last stack map
+        // Note we may be defining a bogus temp for second half of long or double.
+        if (byte_code_offset == -1) {
+          // no stack maps at all, set start to 0
+          byte_code_offset = 0;
+        }
+        new_lvg =
+            mgen.addLocalVariable(
+                "DaIkOnTeMp" + offset, Type.OBJECT, offset, il.findHandle(byte_code_offset), null);
+        debug_instrument.log(
+            "Added local  %s, %d, %d : %s, %s%n",
+            new_lvg.getIndex(),
+            new_lvg.getStart().getPosition(),
+            il.getEnd().getPosition(),
+            new_lvg.getName(),
+            new_lvg.getType());
+        min_size = Math.min(min_size, Type.OBJECT.getSize());
+      }
+    }
+    return offset + min_size;
+  }
+
+  /**
+   * We need to locate and remember any NEW instructions that create uninitialized objects. Their
+   * offset may be contained in a StackMap entry and will probably need to be updated as we add
+   * instrumentation code. Note that these instructions are fairly rare.
+   *
+   * @param il instruction list to search
+   */
+  protected final void build_unitialized_NEW_map(InstructionList il) {
+
+    uninitialized_NEW_map.clear();
+    il.setPositions();
+
+    for (StackMapEntry smte : stack_map_table) {
+      int frame_type = smte.getFrameType();
+
+      if ((frame_type >= Const.APPEND_FRAME && frame_type <= Const.APPEND_FRAME_MAX)
+          || (frame_type == Const.FULL_FRAME)) {
+
+        if (smte.getNumberOfLocals() > 0) {
+          for (StackMapType smt : smte.getTypesOfLocals()) {
+            if (smt.getType() == Const.ITEM_NewObject) {
+              int i = smt.getIndex();
+              uninitialized_NEW_map.put(il.findHandle(i), i);
+            }
+          }
+        }
+        if (smte.getNumberOfStackItems() > 0) {
+          for (StackMapType smt : smte.getTypesOfStackItems()) {
+            if (smt.getType() == Const.ITEM_NewObject) {
+              int i = smt.getIndex();
+              uninitialized_NEW_map.put(il.findHandle(i), i);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * One of these special NEW instructions has moved. Update its offset in StackMap entries. Note
```

Comment:
This Javadoc seems to have been copied from `update_uninitialized_NEW_offsets`.  Here, "One of these special NEW instructions" doesn't make sense.